modules/cuml/src/umap.cpp

```diff
@@ -0,0 +1,283 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <node_cuda/utilities/error.hpp>
+#include <node_cudf/utilities/napi_to_cpp.hpp>
+#include <node_cuml/cuml/manifold/umap.hpp>
+#include <node_rmm/device_buffer.hpp>
+
+#include <cuml/manifold/umapparams.h>
+#include <cudf/filling.hpp>
+#include <cudf/reduction.hpp>
+#include <cudf/types.hpp>
+#include <cuml/common/device_buffer.hpp>
+
+#include <metrics/trustworthiness_c.h>
+#include <raft/linalg/distance_type.h>
+
+#include <napi.h>
+
+namespace nv {
+
+int get_int(NapiToCPP const& opt, int const default_val) {
+  return opt.IsNumber() ? opt.operator int() : default_val;
+}
+
+bool get_bool(NapiToCPP const& opt, bool const default_val) {
+  return opt.IsBoolean() ? opt.operator bool() : default_val;
+}
+
+float get_float(NapiToCPP const& opt, float const default_val) {
+  return opt.IsNumber() ? opt.operator float() : default_val;
+}
+
+Napi::Function UMAP::Init(Napi::Env const& env, Napi::Object exports) {
+  return DefineClass(env,
+                     "UMAP",
+                     {InstanceAccessor<&UMAP::n_neighbors>("nNeighbors"),
+                      InstanceAccessor<&UMAP::n_components>("nComponents"),
+                      InstanceAccessor<&UMAP::n_epochs>("nEpochs"),
+                      InstanceAccessor<&UMAP::learning_rate>("learningRate"),
+                      InstanceAccessor<&UMAP::min_dist>("minDist"),
+                      InstanceAccessor<&UMAP::spread>("spread"),
+                      InstanceAccessor<&UMAP::set_op_mix_ratio>("setOpMixRatio"),
+                      InstanceAccessor<&UMAP::local_connectivity>("localConnectivity"),
+                      InstanceAccessor<&UMAP::repulsion_strength>("repulsionStrength"),
+                      InstanceAccessor<&UMAP::negative_sample_rate>("negativeSampleRate"),
+                      InstanceAccessor<&UMAP::transform_queue_size>("transformQueueSize"),
+                      InstanceAccessor<&UMAP::a>("a"),
+                      InstanceAccessor<&UMAP::b>("b"),
+                      InstanceAccessor<&UMAP::initial_alpha>("initialAlpha"),
+                      InstanceAccessor<&UMAP::init>("init"),
+                      InstanceAccessor<&UMAP::target_n_neighbors>("targetNNeighbors"),
+                      InstanceAccessor<&UMAP::target_weight>("targetWeight"),
+                      InstanceAccessor<&UMAP::random_state>("randomState"),
+                      InstanceAccessor<&UMAP::verbosity>("verbosity"),
+                      InstanceAccessor<&UMAP::target_metric>("targetMetric"),
+                      InstanceMethod<&UMAP::fit>("fit"),
+                      InstanceMethod<&UMAP::transform>("transform")});
+}
+
+ML::UMAPParams update_params(NapiToCPP::Object props) {
+  auto params                  = new ML::UMAPParams();
```

Comment:
Using the new keyword here and never freeing is a memory leak :sweat_smile:. Allocate on the stack and return a copy instead.

```suggestion
  ML::UMAPParams params{};
```